server/replication/replication_mode.go

```diff
@@ -209,6 +224,23 @@ func (m *ModeManager) loadDRAutoSync() error {
 	return nil
 }
 
+func (m *ModeManager) drCheckAsyncTimeout() bool {
+	m.RLock()
+	defer m.RUnlock()
+	timeout := m.config.DRAutoSync.WaitAsyncTimeout.Duration
+	if timeout == 0 {
+		return true
+	}
+	// make sure all members are timeout.
+	for _, t := range m.drMemberWaitAsyncTime {
+		if time.Since(t) <= timeout {
+			return false
+		}
+	}
+	// make sure all members that have synced with previous leader are timeout.
+	return time.Since(m.initTime) > timeout
```

Comment:
If the pd leader is started at `0:00` and the follower is started at `0:03`, and the `WaitAsyncTimeout` is 2min. If the `drCheckAsyncTimeout` happened in the `2:01` and the follower's request arrive at the `2:02`. The logic here will judge that all the members are timeout. Is this expeceted?

Reply:
Should be ok. In order to determine if data is synced, we need to check if `time.Now() < lastSync.Add(2min)` too.

Reply:
got it